UnityAds/src/main/java/com/mopub/mobileads/UnityBanner.java

```diff
@@ -132,51 +118,20 @@ public void onUnityBannerClick(String placementId) {
     }
 
     @Override
-    public void onUnityBannerHide(String placementIds) {
-        MoPubLog.log(CUSTOM, ADAPTER_NAME, String.format("Banner did hide for placement %s", placementIds));
-    }
-
-    @Override
-    public void onUnityBannerError(String message) {
+    public void onBannerFailedToLoad(BannerView bannerView, BannerErrorInfo errorInfo) {
         MoPubLog.log(CUSTOM, ADAPTER_NAME, String.format("Banner did error for placement %s with error %s",
-                placementId, message));
+                placementId, errorInfo.errorMessage));
 
         if (customEventBannerListener != null) {
```

Comment:
We should adapt this to check if our error code is a no fill otherwise return a different error code

Reply:
And we should always return an error if we had an error not just in no fill. Otherwise we will time out in the waterfall causing us to perform poorly.